processor/k8sprocessor/kube/client.go

```diff
@@ -83,6 +86,7 @@ func New(logger *zap.Logger, apiCfg k8sconfig.APIConfig, rules ExtractionRules,
 	}
 
 	c.informer = newInformer(c.kc, c.Filters.Namespace, labelSelector, fieldSelector)
+	c.namespaceInformer = newSharedNamespaceInformer(c.kc, c.Filters.Namespace, labelSelector, fieldSelector)
```

Comment:
We should not enable this by default. We should check if the user has configured the processor to extract namespace metadata and only initialize the informer if it is needed. The processor should have flag that is set to `true` when namespace metadata extraction is enabled. It'll be useful in other methods to prevent unnecessary function calls and map lookups.